### PR TITLE
Update server_transform.c

### PR DESCRIPTION
**Pull request by dariodip**
Using TSIOBufferWaterMarkSet you can obtain a significant improvement concerning performance

### CODE REVIEW

**Comment by zwoop** [comment]
Does this same pattern apply to any of the existing plugins ?

**Comment by zwoop** [comment]
[approve ci].

**Comment by SolidWallOfCode** [comment]
Yeah, I was wondering about that. One thing to consider if is tweaking the block size might not yield even better results. I'm not sure why this would be noticeably faster - in the end, the same number of block allocations are done, this (AFAICT) simply front loads that. Ultimately it all goes to `MIOBuffer::append_block`. It might be that the pre-allocation causes fewer calls to `read` while processing the network I/O.

**Comment by dariodip** [comment]
> I'm not sure why this would be noticeably faster - in the end, the same number of block allocations are done, this (AFAICT) simply front loads that. Ultimately it all goes to `MIOBuffer::append_block`. It might be that the pre-allocation causes fewer calls to `read` while processing the network I/O.

I've found this looking at the [official documentation](https://docs.trafficserver.apache.org/en/7.1.x/developer-guide/api/functions/TSIOBufferCreate.en.html):

> The watermark of an TSIOBuffer is the minimum number of bytes of data that have to be in the buffer before calling back any continuation that has initiated a read operation on this buffer. As a writer feeds data into the TSIOBuffer, no readers are called back until the amount of data reaches the watermark. Setting a watermark can improve performance because it avoids frequent callbacks to read small amounts of data.

I have also noticed that setting an explicit watermark reduces CPU consuming.

**Comment by bryancall** [comment]
[approve ci]

**Comment by bryancall** [comment]
@ywkaras is going to take a look at this.

**Comment by ywkaras** [comment]
[approve ci fedora]

**Comment by ywkaras** [comment]
[approve ci Fedora]

**Comment by ywkaras** [comment]
Not clear why CI is losing it's mind on this PR.  Maybe it would help to rebase it by hand.

File: example/server_transform/server_transform.c [hunk 1]
```diff
@@ -237,6 +237,7 @@ transform_read(TSCont contp, TransformData *data)
   data->input_reader = NULL;
 
   data->server_vio = TSVConnRead(data->server_vc, contp, data->output_buf, data->content_length);
+  TSIOBufferWaterMarkSet( data->output_buf, data->content_length - sizeof( int ) ); // for performance
```
Comment (ywkaras) [inline example/server_transform/server_transform.c:240]: I think this is wrong.  I think we are still in the process of reading in the full content from the request client.  So the content_length would still be zero here.  It is set after the read from the transform completes:

https://github.com/apache/trafficserver/blob/b6b61fbd3e232dbef010dc8b712bb2076510bd7f/example/plugins/c-api/server_transform/server_transform.c#L151

**Review by ywkaras — CHANGES_REQUESTED** [review]

**Comment by ywkaras** [comment]
This appears to be logically incorrect, author may not be working on TS currently.

**Pull request closed without merging**